custom_components/google_home/api.py

```diff
@@ -417,29 +369,36 @@ async def post(
                         response.status,
                         device.hardware,
                     )
+                    device.available = False
                 else:
                     _LOGGER.error(
                         "Failed to access %s, API returned" " %d: %s",
                         device.name,
                         response.status,
                         response,
                     )
+                    device.available = False
         except ClientConnectorError:
-            _LOGGER.warning(
+            logger_func = _LOGGER.debug if polling else _LOGGER.warning
+            logger_func(
                 "Failed to connect to %s device. The device is probably offline.",
                 device.name,
             )
+            device.available = False
         except ClientError as ex:
             # Make sure that we log the exception from the client if one occurred.
             _LOGGER.error(
-                "Request error: %s",
+                "Request from %s device error: %s",
+                device.name,
                 ex,
             )
+            device.available = False
         except asyncio.TimeoutError:
             _LOGGER.debug(
                 "%s device timed out while trying to post data to it - Raw data: %s",
```

Comment:
Shouldn't it be `%s device timed out while performing an request - Raw data: %s`? 🤔

Reply:
Good catch!